cookbooks/bcpc-hadoop/recipes/datanode.rb

```diff
@@ -1,21 +1,43 @@
 include_recipe 'bcpc-hadoop::hadoop_config'
 include_recipe 'bcpc-hadoop::hive_config'
+::Chef::Recipe.send(:include, Bcpc_Hadoop::Helper)
+Chef::Resource::Bash.send(:include, Bcpc_Hadoop::Helper)
 
 node.default['bcpc']['hadoop']['copylog']['datanode'] = {
     'logfile' => "/var/log/hadoop-hdfs/hadoop-hdfs-datanode-#{node.hostname}.log",
     'docopy' => true
 }
 
-%w{hadoop-yarn-nodemanager
-   hadoop-hdfs-datanode
-   hadoop-mapreduce
-   hadoop-client
-   sqoop
-   lzop
-   cgroup-bin
-   hadoop-lzo}.each do |pkg|
+hdp_select_pkgs = %w{hadoop-yarn-nodemanager
+                     hadoop-hdfs-datanode
+                     hadoop-client
+                     }
+
+(hdp_select_pkgs.map{|p| hwx_pkg_str(p, node[:bcpc][:hadoop][:distribution][:release])} +
+                  %w{hadoop-mapreduce
+                     sqoop
+                     lzop
+                     libmysql-java
+                     cgroup-bin
+                     hadoop-lzo}).each do |pkg|
   package pkg do
-    action :upgrade
+    action :install
+  end
+end
+
+(hdp_select_pkgs + ['sqoop-client', 'sqoop-server']).each do |pkg|
+  bash "hdp-select #{pkg}" do
+    code "hdp-select set #{pkg} #{node[:bcpc][:hadoop][:distribution][:release]}"
+    subscribes :run, "package[#{hwx_pkg_str(pkg, node[:bcpc][:hadoop][:distribution][:release])}]", :immediate
+    action :nothing
+  end
+end
+
+hdp_select_pkgs.each do |pkg|
```

Comment:
Ah; yeah, `hdp-select` has two options for `sqoop` while there is only one package.